src/write.c

```diff
@@ -3226,24 +3232,25 @@ avifResult avifEncoderFinish(avifEncoder * encoder, avifRWData * output)
-    }                                                                      //
-    if (isSequence) {                                                      //
-        AVIF_CHECKRES(avifRWStreamWriteChars(&s, "avis", 4));              // ... compatible_brands[]
-        AVIF_CHECKRES(avifRWStreamWriteChars(&s, "msf1", 4));              // ... compatible_brands[]
-        AVIF_CHECKRES(avifRWStreamWriteChars(&s, "iso8", 4));              // ... compatible_brands[]
-    }                                                                      //
-    AVIF_CHECKRES(avifRWStreamWriteChars(&s, "mif1", 4));                  // ... compatible_brands[]
-    AVIF_CHECKRES(avifRWStreamWriteChars(&s, "miaf", 4));                  // ... compatible_brands[]
-    if ((imageMetadata->depth == 8) || (imageMetadata->depth == 10)) {     //
-        if (imageMetadata->yuvFormat == AVIF_PIXEL_FORMAT_YUV420) {        //
-            AVIF_CHECKRES(avifRWStreamWriteChars(&s, "MA1B", 4));          // ... compatible_brands[]
-        } else if (imageMetadata->yuvFormat == AVIF_PIXEL_FORMAT_YUV444) { //
-            AVIF_CHECKRES(avifRWStreamWriteChars(&s, "MA1A", 4));          // ... compatible_brands[]
+    AVIF_CHECKRES(avifRWStreamWriteChars(&s, majorBrand, 4));          // unsigned int(32) major_brand;
+    AVIF_CHECKRES(avifRWStreamWriteU32(&s, minorVersion));             // unsigned int(32) minor_version;
+    AVIF_CHECKRES(avifRWStreamWriteChars(&s, "avif", 4));              // unsigned int(32) compatible_brands[];
+    if (useAvioBrand) {                                                //
+        AVIF_CHECKRES(avifRWStreamWriteChars(&s, "avio", 4));          // ... compatible_brands[]
+    }                                                                  //
+    if (isSequence) {                                                  //
+        AVIF_CHECKRES(avifRWStreamWriteChars(&s, "avis", 4));          // ... compatible_brands[]
+        AVIF_CHECKRES(avifRWStreamWriteChars(&s, "msf1", 4));          // ... compatible_brands[]
+        AVIF_CHECKRES(avifRWStreamWriteChars(&s, "iso8", 4));          // ... compatible_brands[]
+    }                                                                  //
+    AVIF_CHECKRES(avifRWStreamWriteChars(&s, "mif1", 4));              // ... compatible_brands[]
+    AVIF_CHECKRES(avifRWStreamWriteChars(&s, "miaf", 4));              // ... compatible_brands[]
+    if (isMA1B) {                                                      //
+        AVIF_CHECKRES(avifRWStreamWriteChars(&s, "MA1B", 4));          // ... compatible_brands[]
+    }                                                                  //
+    if ((imageMetadata->depth == 8) || (imageMetadata->depth == 10)) { //
+        if (imageMetadata->yuvFormat == AVIF_PIXEL_FORMAT_YUV444) {    //
+            AVIF_CHECKRES(avifRWStreamWriteChars(&s, "MA1A", 4));      // ... compatible_brands[]
```

Comment:
Are the constraints listed in https://aomediacodec.github.io/av1-avif/v1.1.0.html#advanced-profile always true here?  
Note that they differ for items and sequences.